src/otx/algorithms/common/adapters/mmcv/utils/_builder_build_data_parallel.py

```diff
@@ -174,6 +174,7 @@ def __init__(self, *args, enable_autocast: bool = False, **kwargs):
         super().__init__(*args, **kwargs)
         self.enable_autocast = enable_autocast
         self.src_device_obj = torch.device("hpu", self.device_ids[0])
+        self.module.to(self.src_device_obj)
```

Comment:
We don't need this because `module` is already set to `cuda`, isn't it?

Reply:
In segmentation, it isn't converted to cuda. So we need to move model on proper device somewhere.

Reply:
https://github.com/openvinotoolkit/training_extensions/blob/902f35dc2b86ce57bd5a14f25b55f22e1a008c71/src/otx/algorithms/segmentation/adapters/mmseg/apis/train.py#L77
```python
model = build_dp(model, "cuda" if cfg.device == "hpu" else cfg.device, device_ids=cfg.gpu_ids) 
```
I think we can change like the above instead of changing `HPUDataParallel`, what do you think?
I'm worried about that this part can do duplicated operation for some models, like classification.

Reply:
I'll update it

Reply:
When I checked the code, `cfg.device` should be "hpu" because it's referred to decide which type of DataParallel class to use. So I just add the code `model.to` in `mmseg.api.train.py`